src/utils/label.ts

```diff
@@ -0,0 +1,112 @@
+import * as log from '../log'
+import { store } from '../logic'
+import * as ns from '../ns'
+import * as rdf from 'rdflib' // pull in first avoid cross-refs
+
+const UI = { log, ns, rdf, store }
+
+// This ubiquitous function returns the best label for a thing
+//
+// The hacks in this code make a major difference to the usability
+export function label (thing, initialCap = false): string {
+  function doCap (label: string) {
+    if (initialCap) {
+      return label.slice(0, 1).toUpperCase() + label.slice(1)
+    }
+    return label
+  }
+  function cleanUp (label: string) {
+    let result = ''
+    if (label.slice(-1) === '/') label = label.slice(0, -1) // chop trailing slash
+    for (let i = 0; i < label.length; i++) {
+      if (label[i] === '_' || label[i] === '-') {
+        result += ' '
+        continue
+      }
+      result += label[i]
+      if (
+        i + 1 < label.length &&
+        label[i].toUpperCase() !== label[i] &&
+        label[i + 1].toLowerCase() !== label[i + 1]
+      ) {
+        result += ' '
+      }
+    }
+    if (result.slice(0, 4) === 'has ') result = result.slice(4)
+    return doCap(result)
+  }
+
+  const label = getWellKnownLabel(thing)
+
+  if (label) {
+    return doCap(label.value)
+  }
+
+  // Default to label just generated from the URI
+
+  if (thing.termType === 'BlankNode') {
+    return '...'
+  }
+  if (thing.termType === 'Collection') {
+    return '(' + thing.elements.length + ')'
+  }
+  let s = thing.uri
+  if (typeof s === 'undefined') return thing.toString() // can't be a symbol
+  // s = decodeURI(s) // This can crash is random valid @ signs are presentation
+  // The idea was to clean up eg URIs encoded in query strings
+  // Also encoded character in what was filenames like @ [] {}
+  try {
+    s = s
+      .split('/')
+      .map(decodeURIComponent)
+      .join('/') // If it is properly encoded
+  } catch (e) {
+    // try individual decoding of ASCII code points
+    for (let i = s.length - 3; i > 0; i--) {
+      const hex = '0123456789abcefABCDEF' // The while upacks multiple layers of encoding
+      while (
+        s[i] === '%' &&
+        hex.indexOf(s[i + 1]) >= 0 &&
+        hex.indexOf(s[i + 2]) >= 0
+      ) {
+        s =
+          s.slice(0, i) +
+          String.fromCharCode(parseInt(s.slice(i + 1, i + 3), 16)) +
+          s.slice(i + 3)
+      }
+    }
+  }
+  if (s.slice(-5) === '#this') s = s.slice(0, -5)
+  else if (s.slice(-3) === '#me') s = s.slice(0, -3)
+
```

Comment:
How about:
```
if s.endsWith('/profile/card') {
return s.split('/')[2].split('.')[0] // Use domain name
}
```

Reply:
I kept the slicing for now, and extraced it to a small function at least. I am not totally happy with it but don't have time right now. If you are fine with it feel free to merge, otherwise I will try to find some time for further refactoring later

Reply:
Lets go with this